tests/test_operator.py

```diff
@@ -1375,12 +1371,31 @@ def test_conditional_declarations(self):
         list_initialize = Expression(ClusterizedEq(Eq(a[x], init_value)))
         iet = Conditional(x < 3, list_initialize, list_initialize)
         iet = Callable('test', iet, 'void')
-        iet = DataManager.place_definitions.__wrapped__(DataManager(None, None), iet)[0]
+        iet = CDataManager.place_definitions.__wrapped__(CDataManager(None, None), iet)[0]
         for i in iet.body.body[0].children:
             assert len(i) == 1
             assert i[0].is_Expression
             assert i[0].expr.rhs is init_value
 
+    def test_nested_scalar_assigns(self):
+        grid = Grid(shape=(4, 4))
+
+        f = Function(name='f', grid=grid)
+        s = Symbol(name='s', dtype=grid.dtype)
+
+        eqns = [Eq(s, 0),
+                Eq(s, s + f + 1)]
```

Comment:
So does that mean `norm` can be simplified in our bultins or do we still need that size `1` Function

Reply:
not really, what makes you think that ?